src/graph/graph_traversal.cc

```diff
@@ -0,0 +1,107 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/traversal.cc
+ * \brief Graph traversal implementation
+ */
+#include <dgl/graph_traversal.h>
+#include <dgl/packed_func_ext.h>
+#include "../c_api_common.h"
+
+using namespace dgl::runtime;
+
+namespace dgl {
+namespace traverse {
+
+DGL_REGISTER_GLOBAL("traversal._CAPI_DGLBFSNodes_v2")
+.set_body([] (DGLArgs args, DGLRetValue* rv) {
+    HeteroGraphRef g = args[0];
+    const IdArray src = args[1];
+    bool reversed = args[2];
+    aten::CSRMatrix csr;
+    if (reversed) {
+      csr = g.sptr()->GetCSCMatrix(0);
+    } else {
+      csr = g.sptr()->GetCSRMatrix(0);
+    }
```

Comment:
Please raise a better error message when the graph is restricted from using CSR/CSC.

Reply:
I suppose GetCSCMatrix will give an error.